pkgs/games/heroic/default.nix

```
@@ -1,39 +1,69 @@
-{ lib, fetchurl, appimageTools }:
-
-let
-  pname = "heroic";
-  version = "2.2.6";
-  name = "${pname}-${version}";
-  src = fetchurl {
-    url = "https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/releases/download/v${version}/Heroic-${version}.AppImage";
-    sha256 = "sha256-kL30/G4DpDPwGN7PvbWest7TcgL4Rd1c2OM4nRCT3bg=";
+{ lib
+, mkYarnPackage
+, fetchFromGitHub
+, makeWrapper
+, electron
+
+, autoPatchelfHook
+, stdenv
+, zlib
+}:
+
+mkYarnPackage rec {
+  pname = "heroic-original";
+  version = "2.4.0";
+
+  src = fetchFromGitHub {
+    owner = "Heroic-Games-Launcher";
+    repo = "HeroicGamesLauncher";
+    rev = "v${version}";
+    sha256 = "sha256-x0aJIsoKYo0aBtxglZpT5EMgLTqSekf+noEBq/0Wq2s=";
   };
-  appimageContents = appimageTools.extractType2 { inherit name src; };
 
-in
-appimageTools.wrapType2 {
-  inherit name src;
+  packageJSON = ./package.json;
+  yarnLock = ./yarn.lock;
+  yarnNix = ./yarn.nix;
 
-  extraInstallCommands = ''
-    mv $out/bin/${name} $out/bin/${pname}
+  nativeBuildInputs = [
+    makeWrapper
+    autoPatchelfHook
+  ];
 
-    mkdir -p $out/share/${pname}
-    cp -a ${appimageContents}/locales $out/share/${pname}
-    cp -a ${appimageContents}/resources $out/share/${pname}
+  extraBuildInputs = [
+    zlib
+    stdenv.cc.cc # libstdc++.so.6
+  ];
+
+  DISABLE_ESLINT_PLUGIN = "true";
+
+  postBuild = let
+    yarnCmd = "yarn --production --offline --frozen-lockfile --ignore-engines --ignore-scripts --lockfile ${yarnLock}";
+  in ''
+    ${yarnCmd} build-electron
+    ${yarnCmd} build
+  '';
 
-    install -m 444 -D ${appimageContents}/heroic.desktop -t $out/share/applications
+  doDist = false;
 
-    cp -a ${appimageContents}/usr/share/icons $out/share/
+  postInstall = let
+    deps = "$out/libexec/heroic/deps/heroic";
+  in ''
+    makeWrapper "${electron}/bin/electron" "$out/bin/heroic" \
+      --append-flags "${deps}/build/main.js" \
+      --prefix PATH : "${deps}/build/linux"
 
-    substituteInPlace $out/share/applications/heroic.desktop \
-      --replace 'Exec=AppRun' 'Exec=heroic'
+    substituteInPlace "${deps}/flatpak/com.heroicgameslauncher.hgl.desktop" \
+      --replace "Exec=heroic-run" "Exec=heroic"
+    mkdir -p "$out/share/applications" "$out/share/icons/hicolor/512x512/apps"
+    ln -s "${deps}/flatpak/com.heroicgameslauncher.hgl.desktop" "$out/share/applications"
+    ln -s "${deps}/flatpak/com.heroicgameslauncher.hgl.png" "$out/share/icons/hicolor/512x512/apps"
   '';
 
   meta = with lib; {
-    description = "A Native GUI Epic Games Launcher for Linux, Windows and Mac";
+    description = "A Native GOG and Epic Games Launcher for Linux, Windows and Mac";
     homepage = "https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher";
     license = licenses.gpl3Only;
-    maintainers = with maintainers; [ wolfangaukang ];
+    maintainers = with maintainers; [ wolfangaukang aidalgol ];
```

```suggestion
    maintainers = with maintainers; [ aidalgol ];
```

As the whole package is being restructured, I'd rather have myself removed from the maintainers list.

---

I'm quite happy to take over looking after this package.  Thank you very much for your initial work in packing this for NixOS. :slightly_smiling_face:

---

Sure thing, in fact I would thank you for the effort of complementing this package with a decent wrapper. I do not play games on my PC anymore, so I can't find any other reason to maintain this package.